src/CharterManager.sol

```diff
@@ -195,34 +212,49 @@ contract CharterManagerImp {
         ManagedGemJoinLike(gemJoin).exit(urp, usr, val);
     }
 
+    function draw(bytes32 ilk, address u, address urp, address w, int256 dink, int256 dart, uint256 rate, uint256 _gate) internal {
```

Comment:
This is probably a quite opinionated comment, but still want to at least leave the comment (feel free to discard it).
I think these two functions `draw` and `validate` should just be part of the `frob` function like they were before.
I don't see it is necessary to move them to separate `internal` functions when they are used just once.
Even if the function gets a bit long, I think it would be even clearer just make the whole flow go in one function.

Reply:
As we talked we will leave it with the internal functions.
Did not find a nice enough was to do it all in one function and not re-read storage variables redundantly.